### PR TITLE
Gorilla mux

### PR DESCRIPTION
**Pull request by Tejpald3v**
Please review and suggest me on following:
1. In which area I have to improve more.
2. Anything that I am doing bad or bad programming practice.
3. And if you could suggest what further steps I should take.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -1,9 +1,57 @@
...
+	s := http.Server{
+		Addr:         "127.0.0.1:8080",  // configure the bind address
+		Handler:      r,                 // set the default handler
+		ReadTimeout:  5 * time.Second,   // max time to read request from the client
+		WriteTimeout: 10 * time.Second,  // max time to write response to the client
+		IdleTimeout:  120 * time.Second, // max time for connections using TCP Keep-Alive
+	}
+
+	// start the server
+	go func() {
```
Comment (ashokhirpara1) [inline main.go:35]: We don't need this go func and shutdown code here .. simple listenandserve will work

File: routes/defaulthttp/default_handler.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+	var u User
+	err := json.NewDecoder(r.Body).Decode(&u)
+	if err != nil {
+		http.Error(rw, err.Error(), 400)
+		log.Fatal(err)
+	}
+	return u
+}
+
+func writeResponse(rw http.ResponseWriter, s User) {
```
Comment (ashokhirpara1) [inline routes/defaulthttp/default_handler.go:33]: Please use this `writeResponse ` function for all your http responses, weather it is success/error and data

File: routes/gorilla/routes.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+	putRouter := r.Methods(http.MethodPut).Subrouter()
+	putRouter.HandleFunc("/put/{id}", Put)
+	putRouter.Use(checkID)
+
+	postRouter := r.Methods("POST").Subrouter()
+	postRouter.HandleFunc("/post", Post)
+
+	deleteRouter := r.Methods("DELETE").Subrouter()
+	deleteRouter.HandleFunc("/delete/{id}", Delete)
+	putRouter.Use(checkID)
```
Comment (ashokhirpara1) [inline routes/gorilla/routes.go:25]: I think you forgot to change it to `deleteRouter.Use....`

File: routes/defaulthttp/default_handler.go (hunk 2 above)
Comment (Tejpald3v) [reply routes/defaulthttp/default_handler.go:33]: I thought of doing it but was unable to do it before hand there is type miss-match, like I am sending response as message sometimes and struct sometimes which is converted to JSON. How can I do this, can you help me with this?

Comment (ashokhirpara1) [reply routes/defaulthttp/default_handler.go:33]: Check this file, https://github.com/ashokhirpara1/go-jwt-auth/blob/master/routes/responses.go and if you still don't understand then let me know i will explain you.

**Review by ashokhirpara1 — COMMENTED** [review]
@Tejpald3v If you are done with changes then you can merge this PR, Thanks

**Pull request merged**